### ISSUE TITLE
Do you have any plans to make the applicaiton to analyze remote (or online) git repos?

### ISSUE THREAD

**Issue by huskier** [lsk569937453/git-statistic#24]
I understand git-statistic is an offline application, which currently can analyze local git repo. I wonder you have any plans to make the applicaiton to analyze remote (or online) git repos? Analyzing remote git repos will make the application more powerful!

**Comment by lsk569937453** [comment]
You're absolutely right—adding the ability to analyze remote Git repositories would significantly enhance git-statistic. However, platforms like GitHub, Bitbucket, and GitLab each provide different  APIs, which introduces some complexity. If we move forward with this feature, we'll need to prioritize which platform to support first. Personally, I’d like to start with GitHub. Do you have any suggestions or ideas on how best to approach this?

**Comment by huskier** [comment]
We can firstly clone the remote git repos into temporary directories or memory file systems, then they become local git repos.

There are some points to pay attention to:
1. The size of the remote repos;
2. The speed of the network.
